tests/release/e2e-test-default-with-deployment.go

```diff
@@ -0,0 +1,231 @@
+package release
+
+import (
+	"os"
+	"strings"
+
+	"github.com/google/uuid"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/redhat-appstudio/e2e-tests/pkg/constants"
+	"github.com/redhat-appstudio/e2e-tests/pkg/framework"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils/tekton"
+	"knative.dev/pkg/apis"
+
+	ecp "github.com/hacbs-contract/enterprise-contract-controller/api/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	klog "k8s.io/klog/v2"
+)
+
+var _ = framework.ReleaseSuiteDescribe("[HACBS-1199]test-release-e2e-with-deployment", Label("release", "withDeployment", "HACBS"), func() {
+	defer GinkgoRecover()
+	// Initialize the tests controllers
+	framework, err := framework.NewFramework()
+	Expect(err).NotTo(HaveOccurred())
+
+	var defaultEcPolicy = ecp.EnterpriseContractPolicySpec{}
+	var devNamespace = uuid.New().String()
+	var managedNamespace = uuid.New().String()
+
+	var cm = &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: buildPipelineBundleDefaultName,
+		},
+		Data: map[string]string{
+			"default_build_bundle": buildPipelineBundleDefault,
+		},
+	}
+
+	BeforeAll(func() {
+		kubeController := tekton.KubeController{
+			Commonctrl: *framework.CommonController,
+			Tektonctrl: *framework.TektonController,
+		}
+
+		_, err := framework.CommonController.CreateTestNamespace(devNamespace)
+		Expect(err).NotTo(HaveOccurred(), "Error when creating devNamespace: %v", err)
+		klog.Info("Dev Namespace created: ", devNamespace)
+
+		_, err = framework.CommonController.CreateTestNamespace(managedNamespace)
+		Expect(err).NotTo(HaveOccurred(), "Error when creating managedNamespace: %v", err)
+		klog.Info("Managed Namespace created: ", managedNamespace)
+
+		klog.Infof("Wait until the 'pipeline' SA is created in %s namespace \n", managedNamespace)
+		Eventually(func() bool {
+			sa, err := framework.CommonController.GetServiceAccount("pipeline", managedNamespace)
+			return sa != nil && err == nil
+		}, pipelineServiceAccountCreationTimeout, defaultInterval).Should(BeTrue(), "timed out when waiting for the \"pipeline\" SA to be created")
+
+		sourceAuthJson := utils.GetEnv("QUAY_TOKEN", "")
+		Expect(sourceAuthJson).ToNot(BeEmpty())
+
+		destinationAuthJson := utils.GetEnv(constants.QUAY_OAUTH_TOKEN_RELEASE_DESTINATION, "")
```

Comment:
what is the real purpose of using the different quay token than the default one (`QUAY_TOKEN`)?

can't we update the test in a way that it will work only with quay.io repositories that the account in `QUAY_TOKEN` credentials has permissions for?

Reply:
fixed